js/src/APIClient.js

```diff
@@ -43,12 +42,16 @@ class APIClient {
 
       const data = typeof params === 'string' ? params : queryString.stringify(params);
 
+      const urlRegex = new RegExp(/contract\/docx|contract\/pdf|contract\/json/);
+      const responseType = urlRegex.test(url) ? 'blob' : 'json';
```

Comment:
I'm just unsure whether this needs to be a state variable. Isn't that based on the url being called?

Reply:
thanks for catching. you're right. this messes up return values that aren't json. i'll fix so that it sets responseType to blob only for the new download requests (which is required for those).

Reply:
@adridadou I fixed how we handle the `responseType` property by only including it in the `postCallDetails` and `getCallDetails` objects for the relevant download calls. I tested this with all response types and works fine.